lib/core/test/unit/Cardano/Wallet/DB/SqliteSpec.hs

```diff
@@ -188,17 +189,27 @@ loggingSpec = withLoggingDB $ do
 
     describe "Sqlite observables" $ do
         it "should measure query timings" $ \(db, getLogs) -> do
-            let count = 5
+            let count = 1
             replicateM_ count $ listWallets db
-            -- Commented out until this is fixed:
-            --   https://github.com/input-output-hk/iohk-monitoring-framework/issues/391
-            -- msg <- T.unlines . map snd . logMessages <$> getLogs
-            -- T.unpack msgs `shouldContain` "runQuery monitor works"
+
             msgs <- findObserveDiffs <$> getLogs
             length msgs `shouldBe` count
 
+            fullMsgs <- getLogs
+            show fullMsgs `shouldContain` "runQuery monitor works"
+
+            msg <- T.unlines . map snd . logMessages <$> getLogs
+            T.unpack msg `shouldContain` "runQuery monitor works"
+
 newMemoryDBLayer :: IO (DBLayer IO (SeqState DummyTarget) DummyTarget, TVar [LogObject Text])
 newMemoryDBLayer = do
+    logConfig <- testingLogConfig
+    logs <- newTVarIO []
+    (_, db) <- newDBLayer logConfig (traceInTVarIO logs) Nothing
```

Comment:
`traceInTVarIO` will just put every `LogObject` in the `TVar`. It will not simulate all features of the `SwitchBoard` such as monitoring. So I added the spawning of a real `SwitchBoard` which monitors and aggregates.
`commit dda306464871e9fc901226e563cbd1e411d25625
Author: Andreas Triantafyllos <andreas.triantafyllos@iohk.io>
Date:   Fri Jul 5 19:08:51 2019 +0300

    WIP fix monitoring

diff --git a/lib/core/test/unit/Cardano/Wallet/DB/SqliteSpec.hs b/lib/core/test/unit/Cardano/Wallet/DB/SqliteSpec.hs
index 82e8d885..dc5b0b92 100644
--- a/lib/core/test/unit/Cardano/Wallet/DB/SqliteSpec.hs
+++ b/lib/core/test/unit/Cardano/Wallet/DB/SqliteSpec.hs
@@ -19,8 +19,18 @@ import Cardano.BM.Data.MonitoringEval
     ( MEvAction (..), MEvExpr (..), Operand (..), Operator (..) )
 import Cardano.BM.Data.Observable
     ( ObservableInstance (..) )
+import Cardano.BM.Data.Output
+    ( ScribeDefinition (..)
+    , ScribeFormat (..)
+    , ScribeKind (..)
+    , ScribePrivacy (..)
+    )
 import Cardano.BM.Data.Severity
     ( Severity (..) )
+import Cardano.BM.Data.Tracer
+    ( traceWith )
+import Cardano.BM.Setup
+    ( withTrace )
 import Cardano.BM.Trace
     ( traceInTVarIO )
 import Cardano.Wallet.DB
@@ -67,7 +77,7 @@ import Cardano.Wallet.Primitive.Types
 import Cardano.Wallet.Unsafe
     ( unsafeRunExceptT )
 import Control.Monad
-    ( replicateM_, unless )
+    ( forM_, replicateM_, unless )
 import Control.Monad.Trans.Except
     ( runExceptT )
 import Crypto.Hash
@@ -191,7 +201,10 @@ loggingSpec = withLoggingDB $ do
         it "should measure query timings" $ \(db, getLogs) -> do
             let count = 1
             replicateM_ count $ listWallets db
-
+            -- Commented out until this is fixed:
+            --   https://github.com/input-output-hk/iohk-monitoring-framework/issues/391
+            _msgs <- T.unlines . map (T.pack . show) {-. map snd . logMessages-} <$> getLogs
+            T.unpack _msgs `shouldContain` "runQuery monitor works"
             msgs <- findObserveDiffs <$> getLogs
             length msgs `shouldBe` count
 
@@ -206,20 +219,31 @@ newMemoryDBLayer = do
     logConfig <- testingLogConfig
     logs <- newTVarIO []
     (_, db) <- newDBLayer logConfig (traceInTVarIO logs) Nothing
+    withTrace logConfig "" $ \tr -> do
+        logObjs <- readTVarIO logs
+        -- forM_ logObjs $ print
+        forM_ logObjs $ traceWith tr
     pure (db, logs)
 
 testingLogConfig :: IO CM.Configuration
 testingLogConfig = do
     logConfig <- defaultConfigTesting
     CM.setMinSeverity logConfig Debug
-    CM.setSetupBackends logConfig [CM.KatipBK, CM.AggregationBK, CM.MonitoringBK]
+    CM.setSetupBackends   logConfig [CM.KatipBK, CM.AggregationBK, CM.MonitoringBK]
+    CM.setDefaultBackends logConfig [CM.KatipBK]
 
     CM.setSubTrace logConfig "query"
         (Just $ CM.ObservableTraceSelf [MonotonicClock])
 
-    CM.setBackends logConfig
-        "query"
-        (Just [CM.AggregationBK])
+    CM.setSetupScribes logConfig [ ScribeDefinition {
+                                        scName = "out.txt"
+                                      , scKind = FileSK
+                                      , scFormat = ScText
+                                      , scPrivacy = ScPublic
+                                      , scRotation = Nothing
+                                      }
+                                 ]
+
     CM.setAggregatedKind logConfig
         "query"
         (Just CM.StatsAK) -- statistics AgreggatedKind
@@ -231,7 +255,7 @@ testingLogConfig = do
     CM.setMonitors logConfig $ HM.singleton
         "query.diff"
         ( Nothing
-        , Compare "query.diff.timestamp" (GE, (OpMeasurable (CM.Seconds 0)))
+        , Compare "query.diff.Clock.monoclock" (GE, (OpMeasurable (CM.Microseconds 0)))
         , [CreateMessage Info "runQuery monitor works"]
         )
 
`

Reply:
However this is not working too. When Switchboard is about to write to the file or `stdout` it hangs forever.